packages/blockchain-extension/extension/commands/packageSmartContractCommand.ts

```diff
@@ -115,25 +116,37 @@ export async function packageSmartContract(workspace?: vscode.WorkspaceFolder, o
             // Determine the path argument.
             let contractPath: string = workspace.uri.fsPath; // Workspace path
             if (language === 'golang') {
-
                 if (!process.env.GOPATH) {
-                    // The path is relative to $GOPATH/src for Go smart contracts.
-                    const srcPath: string = path.join(contractPath, '..', '..', 'src');
-                    contractPath = path.basename(contractPath);
-                    const exists: boolean = await fs.pathExists(srcPath);
-
-                    if (!exists) {
-                        // Project path is not under GOPATH.
+                    // The path is relative to $GOPATH/src for Go smart contracts.
+                    const indexSrc: number = contractPath.indexOf(path.sep + 'src' + path.sep);
+                    if (indexSrc === -1) {
+                        // Project path is not under GOPATH.
                         throw new Error('The environment variable GOPATH has not been set, and the extension was not able to automatically detect the correct value. You cannot package a Go smart contract without setting the environment variable GOPATH.');
                     } else {
+                        const srcPath: string = contractPath.substring(0, indexSrc + 4);
+                        contractPath = path.relative(srcPath, contractPath);
                         process.env.GOPATH = path.join(srcPath, '..');
                     }
                 } else {
-                    // The path is relative to $GOPATH/src for Go smart contracts.
-                    const srcPath: string = path.join(process.env.GOPATH, 'src');
+                    // The path is relative to $GOPATH/src for Go smart contracts.
+                    const indexSrc: number = contractPath.indexOf(path.sep + 'src' + path.sep);
+                    if (indexSrc === -1) {
+                        throw new Error('The Go smart contract is not a subdirectory of the path specified by the environment variable GOPATH. Please correct the environment variable GOPATH.');
```

Comment:
We are throwing exactly the same error below, ideally we don't want to throw the same error in two different places if possible to avoid.
The code below is already throwing the error under four different conditions, it should be easy to add this one and make it so that we only have the error coming from one location in the code.